homeassistant/components/sabnzbd/__init__.py

```diff
@@ -69,42 +96,79 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
+@callback
+def async_get_entry_id_for_service_call(hass: HomeAssistant, call: ServiceCall) -> str:
+    """Get the entry ID related to a service call (by device ID)."""
+    api_key = call.data[ATTR_API_KEY]
+
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.title == api_key:
```

Comment:
Entry title can be changed by the user. We can't rely on that. We need to use the config entry data.

Reply:
entry is derived from API_KEY entry
https://github.com/home-assistant/core/blob/cdafbbe10f723adf6f8dc2a7db22a7b390d4e6bc/homeassistant/components/sabnzbd/config_flow.py#L60-L63

Reply:
Yes, but the title can be changed later by the user.

Reply:
Using the API key